lib/src/processors/android/icons/android_icons_processor.dart

```diff
@@ -25,27 +25,57 @@
 
 import 'package:flutter_flavorizr/src/extensions/extensions_map.dart';
 import 'package:flutter_flavorizr/src/parser/models/flavorizr.dart';
+import 'package:flutter_flavorizr/src/processors/android/icons/android_adaptive_icon_processor.dart';
+import 'package:flutter_flavorizr/src/processors/android/icons/android_adaptive_icon_xml_processor.dart';
 import 'package:flutter_flavorizr/src/processors/android/icons/android_icon_processor.dart';
 import 'package:flutter_flavorizr/src/processors/commons/queue_processor.dart';
 
 class AndroidIconsProcessor extends QueueProcessor {
   AndroidIconsProcessor({
     required Flavorizr config,
   }) : super(
-          config.androidFlavors
-              .where((_, flavor) =>
-                  flavor.app.icon != null || flavor.android?.icon != null)
-              .map(
-                (flavorName, flavor) => MapEntry(
-                  flavorName,
-                  AndroidIconProcessor(
-                    flavor.android!.icon ?? flavor.app.icon ?? '',
+          [
+            ...config.androidFlavors
+                .where((_, flavor) =>
+                    flavor.app.icon != null || flavor.android?.icon != null)
+                .map(
+                  (flavorName, flavor) => MapEntry(
                     flavorName,
-                    config: config,
+                    AndroidIconProcessor(
+                      flavor.android!.icon ?? flavor.app.icon ?? '',
+                      flavorName,
+                      config: config,
+                    ),
                   ),
-                ),
-              )
-              .values,
+                )
+                .values,
+            ...config.androidFlavors
+                .where((_, flavor) => flavor.android!.adaptiveIcon != null)
+                .map(
+                  (flavorName, flavor) => MapEntry(
+                    flavorName,
+                    AndroidAdaptiveIconXmlProcessor(
+                      flavorName,
+                      config: config,
+                    ),
+                  ),
+                )
+                .values,
+            ...config.androidFlavors
+                .where((_, flavor) => flavor.android!.adaptiveIcon != null)
+                .map(
+                  (flavorName, flavor) => MapEntry(
+                    flavorName,
+                    AndroidAdaptiveIconProcessor(
+                      flavor.android!.adaptiveIcon!.foreground ?? '',
+                      flavor.android!.adaptiveIcon!.background ?? '',
```

Comment:
You can get rid of the null-coalescing operator when you change the `background` and `foreground` attributes to non-null.